portion/dict.py

```diff
@@ -243,57 +230,74 @@ def combine(self, other, how):
 
         return IntervalDict(new_items)
 
+    def as_dict(self):
+        """
+        Return the content as a classical Python dict.
+
+        :return: a Python dict.
+        """
+        return dict(self._storage)
+
+    def __getitem__(self, key):
+        if isinstance(key, Interval):
+            items = []
+            for i, v in self._storage.items():
+                intersection = key & i
+                if not intersection.empty:
+                    items.append((intersection, v))
+            return IntervalDict(items)
```

Comment:
This is a **detail**: a comprehension would certainly be preferable in such a situation:
```suggestion
            return IntervalDict(
                (intersection, v)
                for i, v in self._storage.items()
                for intersection in (key & i,)
                if not intersection.empty
            )
```

Reply:
I tend to disagree with this one, since a comprehension is less readable in this case. Could you detail why you think it is preferable ?

Reply:
I think readability is mainly a question of habits; I’m used to comprehensions; so this is equally readable, but I understand you prefer the C-form.
As far as I know, comprehensions are more efficient than extensions approaches (e.g. `.append()`). However this is likely to be not critical here; that’s why I marked this comment as a **detail** ;)